src/components/Api.js

```diff
@@ -0,0 +1,145 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._authorization = headers.authorization;
+  }
+
+  getAppInfo() {
+    return Promise.all([this.getInitialCards(), this.getUserInfo()]);
+  }
+
+  updateProfileInfo({ title, description }) {
+    return fetch(`${this._baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: {
+        authorization: this._authorization,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        name: title, // API expects title or name?
+        about: description,
+      }),
+    })
+      .then((res) => {
+        return res.ok ? res.json() : Promise.reject(`Error: ${res.status}`);
+      })
+      .catch((err) => {
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very end of any request and this end is in `index.js`   after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.

Please, remove it from `Api` and add it in `index.js`.

Reply:
Thank you for explaining the purpose of catch, that makes great sense. All catch has been removed from api and added in index.js. Thank you!